heareval/tasks/nsynth_pitch.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+"""
+Pre-processing pipeline for NSynth pitch detection
+"""
+
+import os
+from pathlib import Path
+from functools import partial
+import logging
+
+import luigi
+import pandas as pd
+from slugify import slugify
+
+from heareval.tasks.config import NSynthPitchConfig
+from heareval.tasks.util.dataset_builder import DatasetBuilder
+import heareval.tasks.util.luigi as luigi_util
+
+logger = logging.getLogger("luigi-interface")
+config = NSynthPitchConfig()
+
+
+class ConfigureProcessMetaData(luigi_util.WorkTask):
+    """
+    This config is data dependent and has to be set for each data
+    """
+
+    outfile = luigi.Parameter()
+
+    def requires(self):
+        raise NotImplementedError
+
+    @staticmethod
+    def get_rel_path(root: Path, item: pd.DataFrame) -> str:
+        # Creates the audio relative path for a dataframe item
+        audio_path = root.joinpath("audio")
+        filename = f"{item}.wav"
+        return audio_path.joinpath(filename)
+
+    @staticmethod
+    def slugify_file_name(filename: str) -> str:
+        return f"{slugify(filename)}.wav"
+
+    def get_split_metadata(self, split: str) -> pd.DataFrame:
+        logger.info(f"Preparing metadata for {split}")
+
+        # Loads and prepares the metadata for a specific split
+        split_path = Path(self.requires()[split].workdir).joinpath(f"nsynth-{split}")
+
+        metadata = pd.read_json(split_path.joinpath("examples.json"), orient="index")
+
+        # Filter out pitches that are not within the range of a standard piano
+        metadata = metadata[metadata["pitch"] >= 21]
+        metadata = metadata[metadata["pitch"] <= 108]
```

Comment:
Make these config per @khumairraj's suggestion

Reply:
Yes. Maybe we can define it in the `NsynthPitchConfig` after the `super()`. Basically, everything in the super is something that is required across all datasets. Any config specific to the dataset can be after the super. Mostly such a kind of config attribute will be affecting the `ConfigureProcessMetadata` like this pitch thing.